spec/rspec/mocks/partial_double_spec.rb

```diff
@@ -383,6 +383,22 @@ def defined_private_method
         object.implemented
       end
 
+      it 'wraps proxies in a reentrant mutex so we dont trigger deadlocks' do
+        klass = Class.new { def message; end; }
+        called_for = []
+        RSpec.configuration.mock_with(:rspec) do |config|
+          config.before_verifying_doubles do |ref|
+            unless called_for.include? ref.target
+              called_for << ref.target
+              ::RSpec::Mocks.space.proxy_for(ref.target)
+            end
+          end
+          expect { allow(klass.new).to receive(:message) }.to_not raise_error
+          called_for.clear
+          config.verifying_double_callbacks.clear
+        end
+      end
```

Comment:
It's a bit hard to read this spec with it having no blank lines separating the parts.  Can you reformat it a bit?